src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/BasicSSHUserPrivateKey.java

```diff
@@ -157,16 +153,18 @@ public synchronized List<String> getPrivateKeys() {
             List<String> privateKeys = new ArrayList<>();
             for (String privateKey : privateKeySource.getPrivateKeys()) {
                 try {
-                    if (PuTTYKey.isPuTTYKeyFile(new StringReader(privateKey))) {
-                        // strictly we should be encrypting the openssh version with the passphrase, but
-                        // if the key we pass back does not have a passphrase, then the passphrase will not be
-                        // checked, so not an issue.
-                        privateKeys.add(new PuTTYKey(new StringReader(privateKey),
-                                passphrase == null ? "" : passphrase.getPlainText())
-                                .toOpenSSH());
-                    } else {
-                        privateKeys.add(privateKey.endsWith("\n") ? privateKey : privateKey + "\n");
+                    boolean accepted = false;
+                    for (PrivateKeyReader reader : ExtensionList.lookup(PrivateKeyReader.class)) {
+                        if(reader.accept(privateKey)) {
+                            privateKeys.add(reader.toOpenSSH(privateKey, passphrase));
+                            accepted = true;
+                        }
                     }
+                    if (accepted){
+                        continue;
+                    }
```

Comment:
More simply
```suggestion
                    for (PrivateKeyReader reader : ExtensionList.lookup(PrivateKeyReader.class)) {
                        if(reader.accept(privateKey)) {
                            privateKeys.add(reader.toOpenSSH(privateKey, passphrase));
                            continue;
                        }
                    }
```
right?